CombinatorialGames/Nimber/Simplicity.lean

```diff
@@ -0,0 +1,289 @@
+    (∃ a < b + c, P a) ↔ (∃ a < b, P a) ∨ (∃ a < c, P (b + a)) := by
+  simp_rw [lt_add_iff_lt_or_exists_lt]
+  aesop
+
+theorem le_add_iff_lt_or_exists_le {α : Type*} [Add α] [LinearOrder α] [CanonicallyOrderedAdd α]
+    [AddLeftMono α] [IsLeftCancelAdd α] {a b c : α} :
+    a ≤ b + c ↔ a < b ∨ ∃ d ≤ c, a = b + d := by
+  obtain h | h := lt_or_ge a b
+  · have : a ≤ b + c := h.le.trans (le_self_add ..)
+    tauto
+  · obtain ⟨a, rfl⟩ := exists_add_of_le h
+    simp
+
+theorem forall_le_add {α : Type*} [Add α] [LinearOrder α] [CanonicallyOrderedAdd α]
+    [AddLeftMono α] [IsLeftCancelAdd α] {b c : α} {P : α → Prop} :
+    (∀ a ≤ b + c, P a) ↔ (∀ a < b, P a) ∧ (∀ a ≤ c, P (b + a)) := by
+  simp_rw [le_add_iff_lt_or_exists_le]
+  aesop
+
+theorem exists_le_add {α : Type*} [Add α] [LinearOrder α] [CanonicallyOrderedAdd α]
+    [AddLeftMono α] [IsLeftCancelAdd α] {b c : α} {P : α → Prop} :
+    (∃ a ≤ b + c, P a) ↔ (∃ a < b, P a) ∨ (∃ a ≤ c, P (b + a)) := by
+  simp_rw [le_add_iff_lt_or_exists_le]
+  aesop
+
+namespace Ordinal
+
+theorem div_two_opow_log {o : Ordinal} (ho : o ≠ 0) : o / 2 ^ log 2 o = 1 := by
+  apply le_antisymm
+  · simpa [← succ_one] using div_opow_log_lt o one_lt_two
+  · simpa [← succ_zero] using div_opow_log_pos 2 ho
+
+theorem two_opow_log_add {o : Ordinal} (ho : o ≠ 0) : 2 ^ log 2 o + o % 2 ^ log 2 o = o := by
+  convert div_add_mod .. using 2
+  rw [div_two_opow_log ho, mul_one]
+
+protected theorem mul_two (o : Ordinal) : o * 2 = o + o := by
+  rw [← one_add_one_eq_two, mul_add, mul_one]
+
+theorem lt_mul_iff {a b c : Ordinal} : a < b * c ↔ ∃ q < c, ∃ r < b, a = b * q + r := by
+  obtain rfl | hb₀ := eq_or_ne b 0; simp
+  refine ⟨fun h ↦ ⟨_, (Ordinal.div_lt hb₀).2 h, _, mod_lt a hb₀, (div_add_mod ..).symm⟩, ?_⟩
+  rintro ⟨q, hq, r, hr, rfl⟩
+  apply (add_left_strictMono hr).trans_le
+  simp_rw [← mul_succ]
+  apply mul_le_mul_left'
+  simpa
+
+theorem forall_lt_mul {b c : Ordinal} {P : Ordinal → Prop} :
+    (∀ a < b * c, P a) ↔ ∀ q < c, ∀ r < b, P (b * q + r) := by
+  simp_rw [lt_mul_iff]
+  aesop
+
+theorem exists_lt_mul {b c : Ordinal} {P : Ordinal → Prop} :
+    (∃ a < b * c, P a) ↔ ∃ q < c, ∃ r < b, P (b * q + r) := by
+  simp_rw [lt_mul_iff]
+  aesop
+
+theorem mul_add_lt {a b c d : Ordinal} (h₁ : c < a) (h₂ : b < d) : a * b + c < a * d := by
+  apply lt_of_lt_of_le (b := a * (Order.succ b))
+  · rwa [mul_succ, add_lt_add_iff_left]
+  · apply mul_le_mul_left'
+    rwa [Order.succ_le_iff]
+
+instance : CanonicallyOrderedAdd Ordinal where
+  le_self_add := le_add_right
+
+end Ordinal
+
+namespace Nimber
+
+/-! ### Groups -/
+
+/-- Add two nimbers as ordinal numbers. -/
+scoped notation:65 x:arg "+ₒ" y:arg => ∗(toOrdinal x + toOrdinal y)
+
+/-- A nimber `x` is a group when `Iio x` is closed under addition. Note that `0` is a group under
+this definition. -/
+@[mk_iff]
+structure IsGroup (x : Nimber) where
+  add_lt {y z} (hy : y < x) (hz : z < x) : y + z < x
```

Comment:
You could use a strict implicit binder here, which gets rid of the future explicit notation `refine ⟨@fun y z hy hz ↦ ?_⟩` at the cost of needing to explicitly enumerate `add_lt`'s explicit arguments. I've yet to see explicit binders be (non-obviously) avoided for struct hypothesis situations, so take this suggestion with some suspicion.

```suggestion
  add_lt ⦃y z⦄ (hy : y < x) (hz : z < x) : y + z < x
```

Reply:
I like this. I've even used it for very similar situations in the past, see [`Ordinal.Principal`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/SetTheory/Ordinal/Principal.html#Ordinal.Principal), which is basically the same definition!